themes/default/content/blog/mlops-huggingface-llm-aws-sagemaker-python/index.md

```diff
@@ -0,0 +1,183 @@
+
+[Python]:https://www.python.org
+[Python3]:https://www.python.org/downloads
+[natural language prompt]:https://en.wikipedia.org/wiki/Prompt_engineering
+[Meta AI LlaMa 2]:https://ai.meta.com/llama
+[Hugging Face]:https://huggingface.co
+[NousResearch/Llama-2-7b-chat-hf]:https://huggingface.co/NousResearch/Llama-2-7b-chat-hf
+[AWS CLI]:https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+[AWS Credentials]:https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html
+[Amazon SageMaker]:https://aws.amazon.com/pm/SageMaker
+[tremendous value]:https://blogs.nvidia.com/blog/2023/01/26/what-are-large-language-models-used-for
+[Large Language Models]:https://en.wikipedia.org/wiki/Large_language_model
+[LLM]:https://en.wikipedia.org/wiki/Large_language_model
+Welcome to another installment of [Pulumi Python] #MLOpsChallenge!
+
+In this short tutorial we will deploy a publicly available [Meta AI LlaMa 2] based model from [Hugging Face], on [Amazon SageMaker]. Then we will test it with a [natural language prompt] using a short [Python] script.
+
+It may be tempting to dismiss AI as yet another over hyped and complicated technology, but is it really? What if you could start using this cutting edge technology without the stress? AI/ML models, including [Large Language Models] (LLM) like what we will demonstrate today, offer [tremendous value] to organizations, but only if AI/ML platform resources can be reliably, easily, and cost effectively provisioned and put to work. By utilizing [Infrastructure as Code] (IaC) written as [Pulumi Python] programs, machine learning operations teams benefit from [IaC] making deploying AI applications the easiest part of your AI/ML journey. Benefits of [Pulumi Python] based [IaC] for your AI/ML projects include:
+
+1. Easily create AI/ML infrastructure, together with other supporting infrastructure _in the same Pulumi program_
+2. [Pulumi templates] enhance the infrastructure as code (IaC) development experience by providing a pre-written code framework designed to help you bootstrap new Pulumi projects quickly.
+3. Cloud orchestration written in programming languages like [Python] which is already well known and practiced in the AI/ML industry.
+
+## Pulumi makes AI/ML Easy?
+
+For this exercise we will deploy a [Pulumi Template] from scratch. Starting from the `pulumi new` command, we will initialize a new [Pulumi project and stack]. This will provide everything required to deploy our chosen Hugging Face model as an [Amazon SageMaker Model Endpoint]
+
+You can use the `sagemaker-aws-python` template as a working Python starting point for your own Amazon SageMaker deployments, and customize the model you use, SageMaker configuration, and CloudWatch integration as well, just to scratch the surface. Your only limits are your imagination.
+
+Now let's look at what the `sagemaker-aws-python` template will deploy in this tutorial:
+
+* [AWS IAM Roles]
```

Comment:
I would do this maybe in pargraph form so there is more a relationship between everything. 

The template will deploy a Sagemaker instance. IaM roles are attached so it can do <what>. Then huggingface models are downloaded and then mapped to the instance. CW alarms are added to prevent <over cost charges???>. Then an endpoint is exposed and given to you in the outputs. The end point is serving the model